WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java

```diff
@@ -1005,12 +1007,105 @@ protected void signIn() {
             if (isUsernameEmail()) {
                 startProgress(getActivity().getString(R.string.checking_email));
                 requestWPComEmailCheck();
+            } else if (isWPComDomain(mUsername)) {
+                // If a wpcom domain was entered, check if the subdomain matches an existing username.
+                String maybeUsername = extractWPComSubDomain(mUsername);
+                if (maybeUsername.length() > 0) {
+                    // See if the username exists.
+                    startProgress(getActivity().getString(R.string.checking_username));
+                    requestWPComUsernameCheck(maybeUsername);
+                } else {
+                    // The text that was entered was .wordpress.com or the like.
+                    // Its invalid so just show an error.
+                    showUsernameError(R.string.username_invalid);
+                }
             } else {
                 showPasswordFieldAndFocus();
             }
         }
     }
 
+    /**
+     * Tests the specified string to see if it contains a wpcom subdomain.
+     *
+     * @param string The string to check
+     * @return True if the string contains a wpcom subdomain, else false.
+     */
+    private boolean isWPComDomain(String string) {
+        Matcher matcher = WPCOM_DOMAIN.matcher(string);
+        return matcher.find();
+    }
+
+    /**
+     * Extracts the subdomain from a wpcom domain string.
+     * @param domain A wpcom domain is expected.
+     * @return A wpcom subdomain or an empty string.
+     */
+    private String extractWPComSubDomain(String domain) {
```

Comment:
This method doesn't really reflect it's name since it returns the subdomain for *any* URL rather than just wp.com URLs. How about moving this to `UrlUtils` as a general purpose `getSubDomain()` method?